logic/odmr_logic.py

```diff
@@ -201,12 +207,35 @@ def sv_get_fits(self, val):
 
     def _initialize_odmr_plots(self):
         """ Initializing the ODMR plots (line and matrix). """
-        self.odmr_plot_x = np.arange(self.mw_start, self.mw_stop + self.mw_step, self.mw_step)
+
+
+
+        final_freq_list = []
+        for mw_start, mw_stop, mw_step in zip(self.mw_starts, self.mw_stops, self.mw_steps):
+            freqs = np.arange(mw_start, mw_stop + mw_step, mw_step)
+            final_freq_list.extend(freqs)
+            self.frequency_lists.append(freqs)
+
+        if self.final_freq_list:
+            self.odmr_plot_x = np.array(self.final_freq_list)
+        else:
+            self.odmr_plot_x = np.array(final_freq_list)
```

Comment:
For the the if phrase of a list is causing an error. What do you want to check here, whether the list is empty? In that case you can use "if len(self.final_freq_list)".